appserver/data/data-core/src/main/java/fish/payara/data/core/util/FindOperationUtility.java

```diff
@@ -144,21 +188,24 @@ public static Object processFindByOperation(Object[] args, EntityManager em, Que
 
         dataForQuery.setQueryString(builder.toString());
 
-        //here place to process pagination
+        // Process pagination or execute query
         if (evaluatePages) {
             return processPagination(em, dataForQuery, args, dataForQuery.getMethod(), builder, hasWhere, dataParameter);
         } else {
             if (dataParameter.sortList() != null && !dataParameter.sortList().isEmpty()) {
-                handleSort(dataForQuery, dataParameter.sortList(), builder, 
-                        dataForQuery.getQueryType() == QueryType.FIND, false, 
+                handleSort(dataForQuery, dataParameter.sortList(), builder,
+                        dataForQuery.getQueryType() == QueryType.FIND, false,
                         false, null);
                 dataForQuery.setQueryString(builder.toString());
             }
-            //check order conditions to improve select queries
+
             Query query = em.createQuery(dataForQuery.getQueryString());
+
+            // Set parameters
+            int paramIndex = 1;
             for (int i = 0; i < args.length; i++) {
                 if (!excludeParameter(args[i])) {
-                    query.setParameter(i + 1, args[i]);
+                    query.setParameter(paramIndex++, args[i]);
```

Comment:
Here you are expanding the scope of the variable, if we maintain only the internal i+1, we are creating and deleting the variable until the for cycle finish. just in case you need the variable out of the scope of the for cycle this will make sense